CONTRIBUTING.md

```diff
@@ -0,0 +1,84 @@
+# Contributing to divviup-android
+
+## Prerequisites
+
+Building this project requires a Java JDK (17 or newer), the Android SDK, Rust,
+and Docker to be installed. It is highly recommended to use [Android
+Studio](https://developer.android.com/studio).
+
+### Android Studio
+
+Installing Android Studio will provide both a Java JDK and the Android SDK.
+Android Studio is the recommended way to install and manage Android SDK
+components. Refer to the documentation on [compatible versions of the Android
+Gradle plugin and Android Studio][compat-docs] to determine what version to
+install. The Android Gradle plugin's version is set in `/build.gradle.kts`.
+
+[compat-docs]:
+    https://developer.android.com/studio/releases#android_gradle_plugin_and_android_studio_compatibility
+
+### Android SDK
+
+An Android SDK must be installed, along with several additional components. SDK
+components can be managed from Android Studio via Tools -> SDK Manager, or on
+the command line via `$ANDROID_HOME/cmdline-tools/latest/bin/sdkmanager`.
+
+Install the version of "Android SDK Build-Tools" that matches the `compileSdk`
+version in `/divviup/build.gradle.kts`. Install the version of the NDK that
+matches `ndkVersion` in `/divviup/build.gradle.kts`. Install the "Android
+Emulator".
+
+Additionally, you will need to set up a virtual emulator device and download a
+system image for it. This can be done via Tools -> Device Manager in Android
+Studio or `$ANDROID_HOME/cmdline-tools/latest/bin/avdmanager` on the command
+line. Use a system image for a recent API level and the processor architecture
+matching your host computer for day-to-day development.
+
+When making major FFI changes or adding support for a new architecture, you may
+want to return to the device manager and create new virtual devices with
+different architectures for additional testing. Note that there is a significant
+performance difference between using a processor architecture that matches the
+host (which can be virtualized) versus using a dissimilar processor architecture
+(which must be emulated).
+
```

Comment:
```suggestion

### Docker

If you're using an alternative Docker runtime, such as `podman` or `colima`, set up 
your environment according to the [Testcontainers documentation][testcontainers].

[testcontainers]: https://java.testcontainers.org/supported_docker_environment/

```

Caveat: the suggestion might not fit in correctly, suggest taking it in an editor rather than directly committing it.